pkg/core/native/native_neo.go

```diff
@@ -204,6 +204,32 @@ func (n *NEO) Initialize(ic *interop.Context) error {
 	return nil
 }
 
+// InitializeCache initializes all NEO cache with the proper values from storage.
+// Cache initialisation should be done apart from Initialize because Initialize is
+// called only when deploying native contracts.
+func (n *NEO) InitializeCache(bc blockchainer.Blockchainer, d dao.DAO) error {
+	gpb := n.gasPerBlockChanged.Load()
+	if gpb == nil {
```

Comment:
I don't think you need to check for `gpb`.

Reply:
We need this check to distinguish the case when we already initialized all cached values from (NEO).Initialize method. I can safely remove this check, but at the same time, it can save for us several queries to storage.

Reply:
So do we need to remove it?

Reply:
> when we already initialized all cached values from (NEO).Initialize method

Does it ever happen? You only `Initialize` NEO during block 0 processing and it happens some 70 lines above `bc.contracts.NEO.InitializeCache(bc, bc.dao)`.

Reply:
You're right, I didn't notice that we return after storing genesis block.